tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot

```diff
@@ -14,25 +14,87 @@
 
 *** Settings ***
 Documentation  Test 11-03 - Upgrade-InsecureRegistry
-Suite Setup  Install VIC with version to Test Server  7315  --insecure-registry 10.10.10.10:1234
-Suite Teardown  Clean up VIC Appliance And Local Binary
 Resource  ../../resources/Util.robot
 
 *** Keywords ***
-Get host and path from guestinfo
-    ${host}=  Run  govc vm.info -e -json %{VCH-NAME} | jq -c '.VirtualMachines[0].Config.ExtraConfig[] | select(.Key | . and contains("guestinfo.vice./registry/insecure_registries|0/Host")) .Value'
-    ${path}=  Run  govc vm.info -e -json %{VCH-NAME} | jq -c '.VirtualMachines[0].Config.ExtraConfig[] | select(.Key | . and contains("guestinfo.vice./registry/insecure_registries|0/Path")) .Value'
-    [Return]  ${host}  ${path}
+Setup Test Environment
+    [arguments]  ${insecure_registry}
+    ${handle}  ${docker_daemon_pid}=  Start Docker Daemon Locally  --insecure-registry ${insecure_registry}
+    Setup VCH And Registry  ${insecure_registry}
+    [Return]  ${handle}  ${docker_daemon_pid}
+
+Add Project On Registry
+    [Tags]  secret
+    [Arguments]  ${registry_ip}  ${http}  ${user}=admin  ${password}=%{TEST_PASSWORD}
+    # Harbor API: https://github.com/vmware/harbor/blob/master/docs/swagger.yaml
+    ${rc}=  Run Keyword If  ${http}  Run And Return Rc  curl -u ${user}:${password} -H "Content-Type: application/json" -X POST -d '{"project_name": "test","public": 1}' http://${registry_ip}/api/projects
+    Run Keyword If  ${http}  Should Be Equal As Integers  ${rc}  0
+    ${rc}=  Run Keyword If  not ${http}  Run And Return Rc  curl --insecure -u ${user}:${password} -H "Content-Type: application/json" -X POST -d '{"project_name": "test","public": 1}' https://${registry_ip}/api/projects
+    Run Keyword If  not ${http}  Should Be Equal As Integers  ${rc}  0
+
+Setup VCH And Registry
+    [Tags]  secret
+    [Arguments]  ${registry_ip}  ${registry_user}=admin  ${registry_password}=%{TEST_PASSWORD}
+    ${rc}  ${output}=  Run And Return Rc And Output  echo "From busybox" | docker -H ${default_local_docker_endpoint} build -t ${registry_ip}/test/busybox -
+    Should Be Equal As Integers  ${rc}  0
+    Log To Console  \npulling busybox\n${output}
+    ${rc}  ${output}=  Run And Return Rc And Output  docker -H ${default_local_docker_endpoint} login --username ${registry_user} --password ${registry_password} ${registry_ip}
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Login Succeeded
+    ${rc}=  Run And Return Rc  docker -H ${default_local_docker_endpoint} push ${registry_ip}/test/busybox
+    Should Be Equal As Integers  ${rc}  0
+
+Test VCH And Registry
+    [Arguments]  ${vch_endpoint}  ${registry_ip}
+    ${rc}  ${output}=  Run And Return Rc And Output  docker -H ${vch_endpoint} pull ${registry_ip}/test/busybox
+    Should Be Equal As Integers  ${rc}  0
+    Log To Console  ${output}
```

Comment:
Please remove this as it adds verbosity to the test output. Instead, we can check for the expected output. See https://github.com/vmware/vic/blob/master/tests/resources/Docker-Util.robot#L34